src/Symfony/Component/VarDumper/Resources/functions/dump.php

```diff
@@ -9,28 +9,31 @@
  * file that was distributed with this source code.
  */
 
-use Symfony\Component\VarDumper\Caster\ScalarStub;
 use Symfony\Component\VarDumper\VarDumper;
+use Symfony\Component\VarDumper\Dumper\VarDumperOptions;
 
 if (!function_exists('dump')) {
     /**
      * @author Nicolas Grekas <p@tchwork.com>
      * @author Alexandre Daubois <alex.daubois@gmail.com>
+     *
+     * @return mixed|VarDumperOptions
      */
     function dump(mixed ...$vars): mixed
```

Comment:
For IDE completion, it would be useful to add this return type.
```suggestion
    function dump(mixed ...$vars): mixed|VarDumperOptions
```

Reply:
I wanted to, but PHPStorm told me that mixed type must be used as a standalone type and can't be used in an union type 😕

Reply:
Does `@return mixed|VarDumperOptions` provide the target experience with phpstorm?

Reply:
Could be `@template T` + `@param ...T` + `@return T|VarDumperOptions` work with both suggesting the options + the methods on the type passed as argument?

Reply:
https://psalm.dev/r/6b453ad4c5 :eyes:

Reply:
Thank you both for helping with this, just learnt about template types!

@nicolas-grekas' solution works well, we get autocompletion of both the options and what's passed in argument. @ro0NL I tried to put your solution in PHPStorm, but it seems it doesn't understand this and autocompletion does not work, unfortunately 😕

Reply:
IIUC it's on their list https://youtrack.jetbrains.com/issue/WI-66465/Support-parsing-of-PhpStans-conditional-return-types

Perhaps keep a psalm-return annotations for reference, but ultimately whatever autocompletes in IDE works for me :)